### ISSUE TITLE
Merge `pip_sync.rs` and `pip_install.rs`.

### ISSUE THREAD

**Issue by charliermarsh** [astral-sh/uv#3555]
These used to be more different but they've since converged a lot. The difference now is just that we resolve with --no-deps and we uninstall extraneous packages.

**Comment by palfrey** [comment]
Would this result in `pip sync` gaining various `pip install` args? e.g. `-r foo.txt`

**Comment by zanieb** [comment]
I don't think so, the idea here is that we should combine our internal implementation to avoid duplicated logic.

**Comment by charliermarsh** [comment]
It will enable us to add constraints though.
